### ISSUE TITLE
[V3] Implement new Concept Exercise: nil

### ISSUE THREAD

**Issue by tehsphinx** [exercism/go#1379]
This issue describes how to implement the `nil` concept exercise for the Go track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student about `nil`. It will also introduce the concept of `zero values`.

## Learning objectives

- nil is the empty value for pointers, channels, maps, etc. (complete list of types)
- zero values for string, numbers, bool

## Out of scope

- error checking
- nil checking
- usage of the mentioned types

## Concepts

- `nil` value
- zero values

## Prerequisites

- `types`

## Resources to refer to

- https://golang.org/ref/spec#The_zero_value

### After

- Summary of learning that there is no `empty`/`NULL` value in Go. Go does not have an `un-initialized` state for variables. There are zero-values instead -- a value Go assigns to a variable if no explicit value is specified.
- This also means that a check for `someString == nil` is not possible (or needed).

## Implementing

Suggestion: A series of functions that all have to return an empty value for different types. I would start with some types with zero value `nil` and then move on to other types, so the student (that does not know anything about Go) will try to return `nil` --> learning that some types cannot be `nil`.

This would be the solution:
```go
func EmptySlice() []int {
   return nil
}

func EmptyString() string {
   return ""
}

// etc.
```

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by junedev** [comment]
The topics mentioned in this issues are all covered in the current Concept Exercises.
